data-structures/lists/python/astronaut_list_2D.py

```diff
@@ -0,0 +1,284 @@
+    """Delete a value from an astronaut at the given indices"""
+
+    # Check the astronaut index is valid
+    if astronaut_index >= 0 and astronaut_index < len(astronauts):
+        # Check the value index of the astronaut is valid
+        if value_index >= 0 and value_index < len(astronauts[astronaut_index]):
+            del astronauts[astronaut_index][value_index]
+        else:
+            print(f"Value index {value_index} out of range")
+    else:
+        print(f"Astronaut index {astronaut_index} out of range")
+
+    return astronauts
+
+
+def calculate_total_days(astronauts):
+    """Calculate the sum of all the astronaut's days in space"""
+
+    total_days = 0
+
+    # Repeat for each person in the 2D list
+    for person in astronauts:
+        # Add the number of days to the total
+        total_days += int(person[DAYS_INDEX])
+
+    return total_days
+    
+
+def main():
+    # Data set accurate on Thursday 16th March 2023
+    # Inner list values: craft, name, days in space
+
+    # Initialise the 2D list of astronauts currently in space
+    all_astronauts = [
+        ["ISS", "Sergey Prokopyev", "176"],
+        ["ISS", "Dmitry Petelin", "176"],
+        ["ISS", "Frank Rubio", "176"],
+        ["Shenzhou 15", "Fei Junlong", "112"],
+        ["Shenzhou 15", "Deng Qingming", "107"],
+        ["Shenzhou 15", "Zhang Lu", "107"],
+        ["ISS", "Stephen Bowen", "54"],
+        ["ISS", "Warren Hoburg", "14"],
+        ["ISS", "Sultan Al Neyadi", "14"],
+        ["ISS", "Andrey Fedyaev", "14"]
+    ]
+
+    ### DEMO for the platform ###
+    print("Testing")
+    astronaut_index = 6
+
+    # Output the list for astronaut at given index
+    astronaut = all_astronauts[astronaut_index]
+    print(astronaut)
+
+    # Output name of astronaut at given index
+    name = all_astronauts[astronaut_index][NAME_INDEX]
+    print(name)
+
+
+    ### APPEND NEW ASTRONAUT LIST ###
+    # Append a new astronaut list to the end of the 2D list
+    print("\nAdd a new astronaut to the 2D list")
+
+    # Ask the user to input the new astronaut's details
+    craft = input("Enter the craft's name: ")
+    name = input("Enter the astronaut's full name: ")
+    days = input("Enter the current number of days in space: ")
+
+    # Add the new astronaut to the 2D list
+    all_astronauts = add_new_astronaut(all_astronauts, craft, name, days)
+
+    # Output the list for the new astronaut
+    last_index = len(all_astronauts) - 1
+    astronaut = all_astronauts[last_index]
+    print(astronaut)
+
+
+    ### APPEND NEW VALUE TO ASTRONAUT LIST ###
+    print("\nAdd the country of origin to an astronaut")
+
+    # Ask the user to input the astronaut's index and country of origin
+    index = int(input("Enter the astronaut's index: "))
+    country = input("Enter the astronaut's country of origin: ")
+    
+    # Append the country of origin to the astronaut
+    add_new_value_to_astronaut(all_astronauts, index, country)
```

Comment:
I feel the problem with this is that we know have a longer inner list and the new element is not indexed. We have a very generic add_new_value function whereas I feel if we had an app that allowed this we would just have an empty string for country at outset. It sort of feels that we are mixing up doing something very soecific with doing something more general. I woud go with the specific.